aidial_adapter_bedrock/llm/model/ai21.py

```diff
@@ -0,0 +1,136 @@
+import json
+from typing import Any, Callable, Dict, List, Optional
+
+from pydantic import BaseModel
+
+from aidial_adapter_bedrock.llm.chat_model import PseudoChatModel
+from aidial_adapter_bedrock.llm.consumer import Consumer
+from aidial_adapter_bedrock.llm.model.conf import DEFAULT_MAX_TOKENS_AI21
+from aidial_adapter_bedrock.universal_api.request import ModelParameters
+from aidial_adapter_bedrock.universal_api.token_usage import TokenUsage
+from aidial_adapter_bedrock.utils.concurrency import make_async
+
+
+class TextRange(BaseModel):
+    start: int
+    end: int
+
+
+class GeneratedToken(BaseModel):
+    token: str
+    logprob: float
+    raw_logprob: float
+
+
+class Token(BaseModel):
+    generatedToken: GeneratedToken
+    topTokens: Optional[Any]
+    textRange: TextRange
+
+
+class TextAndTokens(BaseModel):
+    text: str
+    tokens: List[Token]
+
+
+class FinishReason(BaseModel):
+    reason: str  # Literal["length", "endoftext"]
+    length: Optional[int]
+
+
+class Completion(BaseModel):
+    data: TextAndTokens
+    finishReason: FinishReason
+
+
+class AI21Response(BaseModel):
+    id: int
+    prompt: TextAndTokens
+    completions: List[Completion]
+
+    def content(self) -> str:
+        assert (
+            len(self.completions) == 1
+        ), "AI21Response should only have one completion"
+        return self.completions[0].data.text
+
+    def usage(self) -> TokenUsage:
+        assert (
+            len(self.completions) == 1
+        ), "AI21Response should only have one completion"
+        return TokenUsage(
+            prompt_tokens=len(self.prompt.tokens),
+            completion_tokens=len(self.completions[0].data.tokens),
+        )
+
+
+# NOTE: See https://docs.ai21.com/reference/j2-instruct-ref
+def prepare_model_kwargs(model_params: ModelParameters) -> Dict[str, Any]:
+    model_kwargs = {}
+
+    if model_params.max_tokens is not None:
+        model_kwargs["maxTokens"] = model_params.max_tokens
+    else:
+        # The default for max tokens is 16, which is too small for most use cases.
+        # Choosing reasonable default.
+        model_kwargs["maxTokens"] = DEFAULT_MAX_TOKENS_AI21
+
+    if model_params.temperature is not None:
+        model_kwargs["temperature"] = model_params.temperature
+    else:
+        # The default AI21 temperature is 0.7.
+        # The default OpenAI temperature is 1.0.
```

Comment:
I think we need to apply a scale factor. The OpenAI temperature ranges from 0 to 2, whereas in AI21, it ranges from 0 to 1.

Reply:
Indeed. Fixed.